scripts/release-preflight.py

```
@@ -0,0 +1,820 @@
+
+
+def _github_com_environment_with_token(
+    token: str,
+    *,
+    environment: Mapping[str, str],
+) -> dict[str, str]:
+    """Bind an explicit credential to GitHub.com without ambient CLI routing."""
+
+    sanitized = _sanitized_github_cli_environment(environment)
+    if not token or len(token) > 4096 or any(character.isspace() for character in token):
+        raise ReleasePreflightError("explicit GitHub.com token is missing or invalid")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)